typing/parmatch.ml

```diff
@@ -2052,3 +2052,186 @@ let check_partial_param do_check_partial do_check_fragile loc casel =
 let check_partial_gadt pred loc casel =
   check_partial_param (do_check_partial_gadt pred)
     do_check_fragile_gadt loc casel
+
+
+(*************************************)
+(* Ambiguous variable in or-patterns *)
+(*************************************)
+
+module IdSet =
+  Set.Make
+    (struct
+      type t = Ident.t
+      let compare = Pervasives.compare
+    end)
+
+let pattern_vars p = IdSet.of_list (Typedtree.pat_bound_idents p)
+
+(* Row for ambiguous variable search,
+   unseen is the traditional pattern row,
+   seen   is a list of position bindings *)
+
+type amb_row = { unseen : pattern list ; seen : IdSet.t list; }
+
+
+let pretty_set vs =
+  sprintf "{%s}"
+    (String.concat ","
+       (List.map Ident.name (IdSet.elements vs)))
+
+let pretty_seen rs =
```

Comment:
These `pretty_` functions are unused, aren't they?

(I think it is better to have the domain logic in parmatch, and the user-printing logic in the file that knows how to print warnings in general.)